pkg/api/defaults.go

```diff
@@ -377,7 +377,7 @@ func (cs *ContainerService) setOrchestratorDefaults(isUpgrade, isScale bool) {
 		}
 
 		// First, Configure addons
-		cs.setAddonsConfig(isUpdate)
+		cs.setAddonsConfig(isUpgrade)
```

Comment:
was this maybe done to ensure back compat?

Reply:
Passing in upgrade context is indeed to ensure back-compat, and to signal forward-compat transformations. The thing is: back-compat requirements are not a factor for scale operations w/ addons spec configs, because scale is node-only, and addons configurations are control plane-only. See L93 above which sets the `isUpdate` var based on this evaluation:

`isUpdate := isUpgrade || isScale`

What I'm proposing in this PR is that passing `isUpdate` into `setAddonsConfig` is a bug, we should be passing `isUpgrade`.